examples/src/main/java/io/grpc/examples/manualflowcontrol/ManualFlowControlServer.java

```diff
@@ -69,6 +66,11 @@ public void run() {
         }
         final OnReadyHandler onReadyHandler = new OnReadyHandler();
         serverCallStreamObserver.setOnReadyHandler(onReadyHandler);
+        // Set up manual flow control for the request stream. It feels backwards to configure the request
+        // stream's flow control using the response stream's observer, but this is the way it is.
+        serverCallStreamObserver.disableAutoRequest();
```

Comment:
If you set this before setting the onReadyHandler, then you don't need the explicit first request (next line)?

Reply:
Update: take my previous words back. The order of disabling auto request and setting onReadyHandler doesn't matter, they both happen before server starts processing messages. But previously, server side never request the initial message. Why are we going to do this now?

Reply:
The previous release of this code (in v1.29) had an (implicit) `request(1)` for the first message. When it was converted to disableAutoRequestWithInitial() the behavior was changed; to match the old behavior it should have used `disableAutoRequestWithInitial(1)` but it did `disableAutoRequestWithInitial(0)` instead. This was done due to a misunderstanding of the previous behavior.

If the RPC is already ready at this point, will onReady be called? The API isn't required to. The API says it is called when transitioning from false to true; so you technically have to have seen `isReady() == false` before onReady() is guaranteed.

That said, I do know this code currently works without the initial request(), because onReady is called unnecessarily. But I don't want to encourage depending on that behavior.

Reply:
Oh, I'm wrong. The previous implementation didn't do request(1) with manual flow control.